src/classes/CRLP_Batch_Base.cls

```diff
@@ -344,4 +353,195 @@ public abstract class CRLP_Batch_Base {
+
+            if (maxDaysAgo != null && maxYearsAgo != null) {
+                if ((maxDaysAgo / 365) > maxYearsAgo) {
+                    whereClause += '>= LAST_N_DAYS:' + maxDaysAgo;
+                } else {
+                    whereClause += '>= LAST_N_YEARS:' + maxYearsAgo;
+                }
+            } else if (maxDaysAgo != null) {
+                whereClause += '>= LAST_N_DAYS:' + maxDaysAgo;
+            } else if (maxYearsAgo != null) {
+                whereClause += '>= LAST_N_YEARS:' + maxYearsAgo;
+            }
+        }
+        return whereClause += ')';
+    }
+
+    /**
+     * @description Returns true if the org has configured a
+     * user-defined inner join for incremental rollups
+     */
+    private Boolean hasUserDefinedInnerJoin {
+        get {
+            return userDefinedLastNDaysField != null && userDefinedLastNDays != null;
+        }
+    }
+
+    /**
+     * @description Value to use with LAST_N_DAYS: in the inner join query
+     * when the org has the user-defined inner join for incremental rollups
+     */
+    private Integer userDefinedLastNDays {
+        get {
+            if (userDefinedLastNDays == null) {
+                Customizable_Rollup_Settings__c customizableRollupSettings =
+                        UTIL_CustomSettingsFacade.getCustomizableRollupSettings();
+
+                if (customizableRollupSettings.Rollups_IncrementalLastNDays_ValOverride__c != null) {
+                    userDefinedLastNDays = (Integer) customizableRollupSettings.Rollups_IncrementalLastNDays_ValOverride__c;
+                }
+            }
+            return userDefinedLastNDays;
+        }
+        set;
+    }
+
+    /**
+     * @description API name of the date field to use as in the inner join query
+     * when the org has the user-defined inner join for incremental rollups
+     */
+    private String userDefinedLastNDaysField {
+        get {
+            if (userDefinedLastNDaysField == null) {
+                Customizable_Rollup_Settings__c customizableRollupSettings =
+                        UTIL_CustomSettingsFacade.getCustomizableRollupSettings();
+
+                if (String.isNotBlank(customizableRollupSettings.Rollups_IncrementalLastNDays_FldOverride__c)) {
+                    userDefinedLastNDaysField = customizableRollupSettings.Rollups_IncrementalLastNDays_FldOverride__c;
+                }
+            }
+            return userDefinedLastNDaysField;
+        }
+        set;
+    }
+
+    /**
+     * @description Produces the where clause that should be used for the inner join when NOT in incremental mode
+     * @return The WHERE clause, as a string, that should be used to filter the detail records
+     */
+    protected String getStandardWhereClause() {
+        String standardWhereClause = null; //will return null if this method is called from an unsupported rollup type
+        if (jobType == CRLP_RollupProcessingOptions.RollupType.AccountHardCredit
```

Comment:
It's hard to see this correctly in github review, but can you add a blank line befe and after the `else if` blocks in all of these methods to break up the lines a bit. It'll make the methods easier to read.